pkg/kubelet/cm/container_manager.go

```diff
@@ -66,6 +67,9 @@ type ContainerManager interface {
 	// UpdateQOSCgroups performs housekeeping updates to ensure that the top
 	// level QoS containers have their desired state in a thread-safe way
 	UpdateQOSCgroups() error
+
+	// Returns RunContainerOptions with devices, mounts, and env fields populated.
+	GetResources(pod *v1.Pod, container *v1.Container, activePods []*v1.Pod) (*kubecontainer.RunContainerOptions, error)
```

Comment:
Is this method per-pod, or per-container, or per-node? Why not return a set of resources with type that's local to the container manager? Why couple container manager with the `container` package?

Reply:
It is per container. I thought about returning devices, mounts, and envs from this func, but those structs are also part of kubecontainer, so we still need to couple cm with container package. Given that the plan you mentioned to me is to move this function to PodContainerManager which would probably be tied with container package if we move in that direction, I chose to pass the RunContainerOptions struct for future easy extension.
extendability.

Reply:
The description is vague. Can you indicate that this method returns additional resources that needs to be made available for a container in a pod?